rfcs/rfc0019.md

```diff
@@ -0,0 +1,174 @@
+```
+
+will apply qt-like interpolation to TEXT.  This (very rarely-seen) peril will
+exist in qt heredocs as it does in qq heredocs:
+
+```
+$str = <<"END";
+A ${
+END
+}
+END
+```
+
+(The peril here is that the first `END` terminates the string, meaning that the
+`${` is never closed.)
+
+## Backwards Compatibility
+
+Existing static analysis should be able to understand `qt` as a new quote-like
+operator fairly easily.  If the analyzer parses `${...}` within a `qq` string,
+parsing the `{...}` forms in a `qt` string should be possible.
+
+`qt` will need to be made available by a feature guard.  Use of the qt operator
+in source without the feature enabled would often parse as a subroutine call.
+
+There is prior art on the CPAN, most specifically in
+[Quote::Code](https://metacpan.org/pod/Quote::Code), which provides a `qc`.
+The syntax provided by Quote::Code is similar to, but not identical with, that
+proposed here.
+
+## Security Implications
+
+None identified at time of writing.
+
+## Examples
+
+The **Motivation** section, above, provides examples of what *doesn't* work
+well with existing interpolation.  Documentation can start with simple
+interpolation in both, then show how to use qt strings to allow more forms of
+interpolation.
+
+```
+# No interpolation
+qt{Greetings};
+
+# Simple scalar interpolation
+qt<Greetings, {$title} {$name}>;
+
+# Interpolation of method calls
+qt"Greetings, {$user->title} {$user->name}";
+
+# Interpolation of various expressions
+qt{It has been {$since{n}} {$since{units}} since your last login};
+
+qt{...a game of {join q{$"}, $favorites->{game}->name_words->@*}};
+```
+
+## Prototype Implementation
+
+Quote::Code, mentioned above, provides something very similar.
+
+## Future Scope
+
+`qt` is "qq but different".  If successful, it might suggest the usefulness of
+qt-like forms of qr, m, s, and other string-interpolating contexts.  This also
+suggests that another possible design for qt would be a pragma to change the
+behavior of qq *and other interpolating quote-like operators*.  This is more
+complex for the reader (because any given piece of code must be considered in
+terms of the enabled features), but eliminates the proliferation of QLOPs.
+
+## Rejected Ideas
+
+This proposal uses `{...}` instead of `${...}` because `${...}` expects a
+*reference* inside in all other contexts, which is not the case here.
+
+This proposal uses `{...}` (like JavaScript) instead of `#{...}` (like Ruby)
+because the author didn't think the extra character was likely valuable enough
+for disambiguation between literal `{` and start of interpolated expression.
+
+This proposal does not offer a means to interpolate a list without (say)
+`join`.  It would require more syntax and leave the user falling back on
+remembering that multiple arguments to `print` are joined with one thing (`$,`,
+usually an empty string) but multiple list elements interpolated into a string
+are joined with another (`$"`, usually a space).
```

Comment:
FYI I think this is no bad thing. The weird inconsistency between `print @arr` using `$,` but `print "@arr"` using `$"` is already annoying enough and hard to remember. Making the user explicit about what they want here is fine.